### ISSUE TITLE
msiDataObjRsync does not work as expected on iRODS 4.2.0

### ISSUE THREAD

**Issue by cookie33** [irods/irods#3719]
HI,

2 zones of iRODS 4.2.0. In the core.re acSetRescSchemeForCreate and acSetRescSchemeForRepl are set to force the adoption of the default resource, regardless of what the client set. For example:
```
msiSetDefaultResc("cinecaRes1","forced");
```
However the execution of the EUDATReplication rule fails:
```
irule -F eudatReplication2.r ERROR: rcExecMyRule error. status = -24000 SYS_INTERNAL_NULL_INPUT_ERRLevel 0: DEBUG: execMicroService3: error when executing microserviceline 303, col 12, rule base eudat4 msiDataObjRsync(*source,"IRODS_TO_IRODS","null",*destination,*rsyncStatus); ^caused by: DEBUG: msiDataObjRsync: rsDataObjRsync failed for /INGV/home/rods/san/archive/2017/IV/ACER/HHZ.D/IV.ACER..HHZ.D.2017.190, status = -24000
```
And on the target site the error shows that the microservice is trying to create the replica object using the source default resource:
```
/home/irodsbuild/irods/server/core/src/objDesc.cpp:231:int initDataObjInfoWithInp(dataObjInfo_t , dataObjInp_t ) : status [SYS_RESC_DOES_NOT_EXIST] errno [] -- message [] [-] /home/irodsbuild/irods/server/core/src/irods_resource_manager.cpp:1085:irods::error irods::resource_manager::hier_to_leaf_id(const std::string &, rodsLong_t &) : status [SYS_RESC_DOES_NOT_EXIST] errno [] -- message [demoResc]
```

So the microservice msiDataObjRsync does not seem to honor the default resource setting in the core.re. It worked with iRODS v4.1.x.
It is not clear if it is a bug of the microservice or if it depends on some other configuration.

Also note that the icommand irsync works as expected, honouring the default resource settings.

So it is the following command which fails:
```
msiDataObjRsync(*source,"IRODS_TO_IRODS","null",*destination,*rsyncStatus)
```

**Comment by trel** [comment]
Thanks for the report.  We'll try to reproduce this behavior.

**Comment by cookie33** [comment]
HI,

It can be closed. We made an error. The problem was when iRODS 4.2.0 tried to replicate to iRODS 4.1.3.

So it is already fixed in a higher version of iRODS (higher than 4.1.3)

Sorry to bring it up again.

**Comment by trel** [comment]
Ah, very good.   This was the default resource resolution that was fixed in 4.1.9.

https://github.com/irods/irods/issues/2713
